src/weapon.py

```diff
@@ -36,13 +34,15 @@ class Beam(Weapon):
     """
 
     beam_projection: arcade.Sprite
+    button_held: bool
 
     def __init__(self, input_button: VirtualButton, car: arcade.Sprite):
         super().__init__(input_button, car)
         self.beam_projection = arcade.SpriteSolidColor(1000, 5, arcade.color.RED)
+        self.button_held = False
 
     def update(self, delta_time):
-        if not self.button_held and self.input_button.value:
+        if self.input_button.pressed:
             self.shoot()
         if self.button_held:
```

Comment:
Looking at this again, should I change it to `self.input_button.value`?

Reply:
Yeah, definitely. No need to hold up this merge for it though.